modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java

```diff
@@ -329,6 +355,20 @@ private SqlToRelConverter sqlToRelConverter(SqlValidator validator, CalciteCatal
         return new SqlToRelConverter(this, validator, reader, cluster(), convertletTbl, config);
     }
 
+    /** */
+    public void setDisabledRules(Set<String> disabledRuleNames) {
+        ctx.rulesFilter(rulesSet -> {
+            List<RelOptRule> newSet = new ArrayList<>();
+
+            for (RelOptRule r : rulesSet) {
+                if (!disabledRuleNames.contains(r.toString()))
```

Comment:
may be only rule name here would be more clear ? i.e. \w+ only? possibly case insensitive too.

Reply:
Do you propose use regexp patterns?
I guess it is hard to read and may be cause if the performance drop on the planning phase.

Reply:
Withed with substring(). Please take a look